extensions/wikia/Email/Controller/FounderController.class.php

```diff
@@ -0,0 +1,211 @@
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidTitle() {
+		if ( !$this->pageTitle instanceof \Title ) {
+			throw new Check( "Invalid value passed for title" );
+		}
+
+		if ( !$this->pageTitle->exists() ) {
+			throw new Check( "Title doesn't exist." );
+		}
+	}
+
+	/**
+	 * @throws \Email\Check
+	 */
+	private function assertValidRevisionIds() {
+		if ( empty( $this->previousRevId ) ) {
+			throw new Check( "Invalid value passed for previousRevId" );
+		}
+
+		if ( empty( $this->currentRevId ) ) {
+			throw new Check( "Invalid value passed for currentRevId" );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'buttonText' => $this->getChangesLabel(),
+			'buttonLink' => $this->getChangesLink(),
+			'contentFooterMessages' => [
+				$this->getFooterEncouragement(),
+				$this->getFooterArticleLink(),
+				$this->getFooterAllChangesLink(),
+			],
+			'details' => $this->getDetails(),
+			'hasContentFooterMessages' => true
+		] );
+	}
+
+	public function getSubject() {
+		$articleTitle = $this->pageTitle->getText();
+		$name = $this->getCurrentUserName();
+
+		return $this->getMessage( 'emailext-founder-subject', $articleTitle, $name )
+			->text();
+	}
+
+	protected function getSummary() {
+		$articleUrl = $this->pageTitle->getFullURL();
+		$articleTitle = $this->pageTitle->getText();
+
+		return $this->getMessage( 'emailext-founder-summary', $articleUrl, $articleTitle )
+			->parse();
+	}
+
+	protected function getDetails() {
+		$article = \Article::newFromTitle( $this->pageTitle, \RequestContext::getMain() );
+		$service = new \ArticleService( $article );
+		$snippet = $service->getTextSnippet();
+
+		return $snippet;
+	}
+
+	protected function getChangesLabel() {
+		return $this->getMessage( 'emailext-founder-link-label' )->parse();
+	}
+
+	protected function getChangesLink() {
+		return $this->pageTitle->getFullURL( [
+			'diff' => $this->currentRevId,
+			'oldid' => $this->previousRevId,
+		] );
+	}
+
+	protected function getFooterMessages() {
+		$parentUrl = $this->pageTitle->getFullURL( 'action=unwatch' );
+		$parentTitleText = $this->pageTitle->getPrefixedText();
+
+		$footerMessages = [
+			$this->getMessage( 'emailext-unfollow-text', $parentUrl, $parentTitleText )
+				->parse()
+		];
+		return array_merge( $footerMessages, parent::getFooterMessages() );
+	}
+
+	protected function getFooterEncouragement() {
+		$name = $this->getCurrentUserName();
+		$profileUrl = $this->getCurrentProfilePage();
+
+		return $this->getMessage( $this->getFooterEncouragementKey(), $profileUrl, $name )
+			->parse();
+	}
+
+	protected function getFooterEncouragementKey() {
+		return 'emailext-founder-encourage';
```

Comment:
We have been overriding functions like this a lot in these templates so each child can return a different key for a common part of the email.  Constants can technically be overridden as well using ```static::CONST_NAME``` but I've never been a huge fan of that as it seems unintuitive.